qa/abstracts/src/main/java/io/camunda/zeebe/process/test/qa/abstracts/jobs/AbstractTimerTest.java

```diff
@@ -0,0 +1,107 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.camunda.zeebe.process.test.qa.abstracts.jobs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
+import io.camunda.zeebe.client.api.response.DeploymentEvent;
+import io.camunda.zeebe.process.test.api.ZeebeTestEngine;
+import io.camunda.zeebe.process.test.assertions.BpmnAssert;
+import io.camunda.zeebe.process.test.filters.RecordStream;
+import io.camunda.zeebe.process.test.qa.abstracts.util.Utilities;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceRecordValue;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public abstract class AbstractTimerTest {
+
+  private static final String RESOURCE = "test_timer_events.bpmn";
+  private static final String PROCESS_ID = "Process_Timer_Test_01";
+
+  private ZeebeClient client;
+  private ZeebeTestEngine engine;
+  private RecordStream recordStream;
+
+  @BeforeEach
+  void deployProcesses() {
+    final DeploymentEvent deploymentEvent = Utilities.deployResource(client, RESOURCE);
+    BpmnAssert.assertThat(deploymentEvent).containsProcessesByResourceName(RESOURCE);
+  }
+
+  @Test
+  void shouldCompareTimersDueDatesCorrectlyForDifferentNowDates() throws Exception {
+    final List<OffsetDateTime> dates =
+        Arrays.asList(
+            OffsetDateTime.of(2023, 10, 5, 15, 50, 0, 0, ZoneOffset.of("+02:00")),
+            OffsetDateTime.of(2023, 10, 31, 0, 0, 0, 0, ZoneOffset.of("+02:00")),
+            OffsetDateTime.of(2023, 10, 31, 23, 0, 0, 0, ZoneOffset.of("+02:00")),
+            OffsetDateTime.of(2023, 10, 31, 23, 59, 0, 0, ZoneOffset.of("+02:00")),
+            OffsetDateTime.of(2023, 10, 31, 23, 59, 59, 0, ZoneOffset.of("+02:00")),
+            OffsetDateTime.of(2023, 12, 31, 23, 59, 59, 0, ZoneOffset.of("+02:00")));
+
+    for (final OffsetDateTime nowDate : dates) {
+      shouldCompareTimersDueDatesCorrectlyForDifferentNowDates(nowDate);
+    }
+  }
+
+  void shouldCompareTimersDueDatesCorrectlyForDifferentNowDates(final OffsetDateTime nowDate)
```

Comment:
🔧 Please rename this method, is not a test so we can use a different name here. Also add `private` modifier